ntoskrnl/ke/amd64/zeropage.S

```diff
@@ -0,0 +1,29 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Fast zeroing of pages
+ * COPYRIGHT:   Copyright 2021 Jérôme Gardou <jerome.gardou@reactos.org>
+ */
+
+#include <asm.inc>
+
+/* FUNCTIONS ****************************************************************/
+.code
+
+PUBLIC KeZeroPages
+KeZeroPages:
+    pxor xmm0, xmm0
+    /* Calculate buffer end */
+    add rdx, rcx
+    /* Loop */
+do_zero:
+    movntps xmmword ptr [rcx], xmm0
```

Comment:
I checked, whether unrolling could improve performance, but turns out, it doesn't seem to make much of a difference. At least not on my AMD machine. It stayed at around 630 cycles per page. RtlZeroMemory was 350 cycles, RtlSecureZeroMemory (using a simple `rep stosq`) was 250!

Reply:
Yeah, my tests (on Intel) never showed any gain from unrolling either.
Btw, modern processors heavily optimize `rep stosb`, so it's typically worth trying that as well.

Reply:
Since we want to zero pages, i.e. we have both aligned pointer and size and a large size, we can just use `rep stosd` and `rep stosq`, which can be written in C with intrinsics and is most likely the fastest on any system. Maybe AVX 512 can do it faster, but SSE seemingly can't.

Reply:
Even if `movntps` is "only" up to par with `stos`, that probably means it's going to make other code on the system noticeably faster by avoiding unnecessary cache evictions, right?

Reply:
Did some better benchmarking:
`` rep movsq`` : 128
`` movaps``  175
`` movnti`` 535
`` movntdq``: 620
`` movntps``: 620

So `` movntps``  and ``movnti``  are much slower than `` rep movsq`` . Does that help with cache? Different comments here https://stackoverflow.com/questions/10274355/cycles-cost-for-l1-cache-hit-vs-register-on-x86, suggest the difference between L1 and L2 is 6-8 cycles. So 6 * 64 = 384, 8 * 64 = 512. Versus 620 - 128 = 492 cycles slower zeroing. But chances are you are evicting cache lines, you don't need anymore. I guess you would need to write a more complex benchmark.

Reply:
My measurements on Intel typically had `movntps` at the top. But zeroing multiple pages at once provided a much more noticeable gain than any optimization of the zeroing function.

I wonder if we could simply make `MiMapPagesInZeroSpace` disable caching on those pages entirely, to avoid the cache evictions, and then just stick with `stos` for simplicity (and apparently on some processors, performance).

Reply:
Let's stick with stosd/stosq